### PR TITLE
Loki: Improve handling of empty responses

### PR DESCRIPTION
**Pull request by gabor**
helps with https://github.com/grafana/grafana/issues/52046

the loki http response parser calls `converter.ReadPrometheusStyleResult` (https://github.com/grafana/grafana/blob/main/pkg/util/converter/prom.go#L27), which can return a nil-pointer. the loki code did not handle this case. 

the linked github-issue describes such a situation.

this pull-request adds code to handle the nil-pointer case.

how to test:
- the new code is covered by a test, so it should be ok
- try to run some generic loki queries, to make sure this did not break something else

(NOTE:  this will not resolve the reported issue (why are we not getting back any data for the log-volume-query), but we should handle the nil-pointer-case.

### CODE REVIEW

**Review by svennergr — APPROVED** [review]
LGTM

File: pkg/tsdb/loki/api.go [hunk 1]
```diff
@@ -163,6 +163,14 @@ func (api *LokiAPI) DataQuery(ctx context.Context, query lokiQuery) (data.Frames
 	iter := jsoniter.Parse(jsoniter.ConfigDefault, resp.Body, 1024)
 	res := converter.ReadPrometheusStyleResult(iter, converter.Options{MatrixWideSeries: false, VectorWideSeries: false})
 
+	if res == nil {
```
Comment (toddtreece) [inline pkg/tsdb/loki/api.go:166]: would it be better if `ReadPrometheusStyleResult` always returned a result instead of a pointer?

Comment (gabor) [reply pkg/tsdb/loki/api.go:166]: yes, i agree. i created https://github.com/grafana/grafana/issues/52430 for that.

File: pkg/tsdb/loki/api.go [hunk 2]
```diff
@@ -163,6 +163,14 @@ func (api *LokiAPI) DataQuery(ctx context.Context, query lokiQuery) (data.Frames
...
+		// it's hard to say if this is an error-case or not.
+		// we know the http-response was a success-response
+		// (otherwise we wouldn't be here in the code),
+		// so we will go with a success, with no data.
```
[the remark this answers is not in the document]
Comment (gabor) [reply pkg/tsdb/loki/api.go:170]: hmm.. i'd say let's go without logging for now. i'm planning to add some way to record this strange response in a separate PR.

**Pull request merged**